### PR TITLE
Tim/switch to nunjucks

### PR DESCRIPTION
**Pull request by tscanlin**
I refactored the dev docs to use nunjucks instead of swig last week.

@nchilada if you don't wanna review this I understand.. its pretty long..

@tomgenoni FYI, I took some notes on what needed to be refactored in order to transition from swig to nunjucks, I'm happy to share that info with you if you're interested.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,9 @@
     "compass-mixins": "^0.12.7",
     "highlight.js": "^8.9.1",
     "js-yaml": "^3.4.3",
+    "marked": "^0.3.5",
+    "nunjucks": "^2.3.0",
+    "nunjucks-markdown": "^1.1.0",
```
Comment (nchilada) [inline package.json]: These could be `devDependencies`, right?  Can we also delete the `swig` dependencies now?

Comment (tscanlin) [inline package.json]: yes, good call :)

File: src/partials/_article-summary.html [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
-  <a href="{{ fileName }}.html">
-    <h2 class="flush--top push--bottom">{{ title }}</h1>
+  <a href="{{ object.fileName }}.html">
+    <h2 class="flush--top push--bottom">{{ object.title }}</h1>
   </a>
-  {% if summary %}
-    <p class="color--base">{{ summary | marked }}</p>
+  {% if object.summary %}
+    <p class="color--base">
+      {% markdown %}
```
Comment (nchilada) [inline src/partials/_article-summary.html]: Is the Markdown engine unchanged from before?

Comment (tscanlin) [inline src/partials/_article-summary.html]: yep, same markdown parser is used

**Comment by nchilada** [comment]
I'm confused why some templates' variables now have to be dereferenced from an `object` variable, whereas other templates' variable are unaffected.  (And obviously, it would be preferable if all templates' variables were unaffected.)  Is this an artifact of how some templates are referenced in src/content?

**Comment by tscanlin** [comment]
@nchilada the reason the object variable is used is because swig allowed us to pass in data to a partial that is included whereas in nunjucks, included partials automatically inherit data from their parent template. I've adopted a convention of defining an 'object' variable that hold all the properties for a given include/partial. Other top-level templates don't need many changes because they aren't included like this. Hope that makes sense.

**Comment by tscanlin** [comment]
@nchilada after I resolve conflicts is it ok if I merge this? Or did you want to review this some more?

**Comment by nchilada** [comment]
Yep, LGTM!

> I've adopted a convention of defining an 'object' variable that hold all the properties for a given include/partial.
Out of curiosity, how/where is this accomplished?  That's the part that confused me the most.

File: src/templates/page-sidebar--wider.html [hunk 3]
```diff
@@ -9,17 +6,13 @@
...
     <div id="content" class="content soft-half--ends max-width--900">
       {% for section in sections %}
-        {% for i, key in siblingData[section] | sort %}
-          {% set object = siblingData[section][key] %}
-          {%
-            include object.partial
-            object
-          %}
+        {% set sectionData = siblingData[section] %}
+        {% for key, object in sectionData | dictsort %}
```
Comment (tscanlin) [inline src/templates/page-sidebar--wider.html]: @nchilada you were curious so here is where object is coming from, there are a few other examples that are done in a similar way.

Comment (nchilada) [inline src/templates/page-sidebar--wider.html]: I see.  Thanks!

**Comment by tscanlin** [comment]
I have now gone through all pages manually to check consistency between current pages and this branch and didn't see any discrepancies. I'm gonna merge this! :)

**Pull request merged**